src/gtk/desktopenv.cpp

```diff
@@ -0,0 +1,51 @@
+////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/desktopenv.cpp
+// Purpose:     wxDesktopEnv
+// Author:      Igor Korot
+// Created:     2019-10-07
+// Copyright:   (c) 2019 wxWidgets development team
+// Licence:     wxWidgets licence
+/////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#include "wx/wx.h"
+
+#include "gio/gio.h"
+#include "wx/desktopenv.h"
+
+// ============================================================================
+// implementation
+// ============================================================================
+
+wxDesktopEnv::wxDesktopEnv() : wxDesktopEnvBase()
+{
+}
+
+bool wxDesktopEnv::MoveToRecycleBin(wxString &path)
+{
+    bool result = false;
+    if( wxDesktopEnvBase::MoveToRecycleBin( path ) )
+    {
+        GError *error = NULL;
+        GFile *f = g_file_new_for_path( path.c_str() );
+        gboolean res = g_file_trash( f, NULL, &error );
+        if( !res || ( error && error->code == G_IO_ERROR_NOT_SUPPORTED ) )
+           wxLogError( _( "Failed to move file '%s' to Recycle Bin" ), path );
+        else
```

Comment:
I would separate the `G_IO_ERROR_NOT_SUPPORTED` case into a separate log error that says the operation is not supported. Currently, it would be hard to tell if it failed due to the trash being unsupported or another issue.